tfx/src/main/scala/com/spotify/noether/tfx/TfmaConverter.scala

```diff
@@ -30,33 +30,62 @@ trait TfmaConverter[A, B, T <: Aggregator[A, B, _]] {
 object TfmaConverter {
 
   implicit val binaryConfusionMatrixConverter
-    : TfmaConverter[BinaryPred, Map[(Int, Int), Long], BinaryConfusionMatrix] =
-    new TfmaConverter[BinaryPred, Map[(Int, Int), Long], BinaryConfusionMatrix] {
-      override def convertToTfmaProto(underlying: BinaryConfusionMatrix)
-        : Aggregator[BinaryPred, Map[(Int, Int), Long], MetricsForSlice] =
-        underlying
-          .andThenPresent { matrix =>
-            MetricsForSlice
-              .newBuilder()
-              .setSliceKey(SliceKey.getDefaultInstance)
-              .putMetrics(
-                "Noether_BinaryConfusionMatrix",
-                MetricValue
-                  .newBuilder()
-                  .setConfusionMatrixAtThresholds(
-                    newBuilder()
-                      .setMatrices(0,
-                                   ConfusionMatrixAtThreshold
-                                     .newBuilder()
-                                     .setThreshold(underlying.threshold)
-                                     //                  .setFalseNegatives(matrix)))
-                                     .build())
-                      .build())
-                  .build()
-              )
-              .build()
-          }
-    }
+  : TfmaConverter[BinaryPred, Map[(Int, Int), Long], BinaryConfusionMatrix] =
+    (underlying: BinaryConfusionMatrix) => underlying
+      .andThenPresent {
+        cm => {
+          val tp = cm.valueAt(1, 1).toDouble
+          val tn = cm.valueAt(0,0).toDouble
+          val fp = cm.valueAt(1, 0).toDouble
+          val fn = cm.valueAt(0, 1).toDouble
+
+          MetricsForSlice.newBuilder()
+            .setSliceKey(SliceKey.getDefaultInstance)
+            .putMetrics("Noether_BinaryConfusionMatrix",
+              MetricValue.newBuilder()
+                .setConfusionMatrixAtThresholds(ConfusionMatrixAtThresholds.newBuilder()
+                  .addMatrices(ConfusionMatrixAtThreshold.newBuilder()
+                    .setThreshold(underlying.threshold)
+                    .setFalseNegatives(fn)
+                    .setFalsePositives(fp)
+                    .setTrueNegatives(tn)
+                    .setTruePositives(tp)
+                    .setPrecision(tp / (tp + fp))
+                    .setRecall(tp / (tp + fn))
+                    .build())
+                  .build())
+                .build())
+            .build()
+        }
+      }
+
+
+  implicit val confusionMatrixConverter
+  : TfmaConverter[Prediction[Int, Int], Map[(Int, Int), Long], ConfusionMatrix] =
+    (underlying: ConfusionMatrix) => underlying
+      .andThenPresent {
```

Comment:
The implementation of this is identical to `BinaryConfusionMatrix` except for the metric name. And I think we can have these both use the same metric name (`Noether_ConfusionMatrix`) since they operate on different `Prediction` types and hence would never appear together in the same `MultiAggregator`. So can we define a `private def denseMatrixToMetric` that can be used in both cases?

Reply:
BinaryConfusionMatrix also sets the `threshold`, but I suppose we could also pass that in as an argument to denseMatrixToMetric

Reply:
Ah true, yeah passing in an `Option[Double]` for the threshold should be fine

Reply:
@andrewsmartin pushed a change for this, PTAL. The implementation is kind of funky looking but I can't think of a better way to do it right now.